### PR TITLE
Add Timm support in ORTModelForImageClassification

### PR DESCRIPTION
**Pull request by mht-sharma**
# What does this PR do?

Adds the inference of Timm models in `Optimum` using `ORTModelForImageClassification`

```python
from urllib.request import urlopen

import timm
import torch
from PIL import Image

from optimum.onnxruntime import ORTModelForImageClassification

model_id = "timm/resnext101_64x4d.c1_in1k"
model = ORTModelForImageClassification.from_pretrained(model_id, export=True)

# get model specific transforms (normalization, resize)
data_config = timm.data.resolve_data_config(pretrained_cfg=model.config.pretrained_cfg)
transforms = timm.data.create_transform(**data_config, is_training=False)

img = Image.open(
    urlopen("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/beignets-task-guide.png")
)
output = model(transforms(img).unsqueeze(0)).logits  # unsqueeze single image into batch of 1
top5_probabilities, top5_class_indices = torch.topk(torch.softmax(output, dim=1) * 100, k=5)

```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/onnxruntime/modeling_ort.py [hunk 1]
```diff
@@ -574,6 +574,58 @@ def _from_transformers(
             provider_options=provider_options,
         )
 
+    @classmethod
+    def _from_timm(
```
Comment (mht-sharma) [inline optimum/onnxruntime/modeling_ort.py:578]: Need Suggestion: 
The method is exact same as `from_transformers`.  Do we wanna keep them separate for easy reading / just use a single method.

File: optimum/modeling_base.py [hunk 2]
```diff
@@ -285,7 +286,27 @@ def _from_transformers(
...
+        cache_dir: Optional[str] = None,
+        subfolder: str = "",
+        local_files_only: bool = False,
+        trust_remote_code: bool = False,
+        **kwargs,
+    ) -> "OptimizedModel":
+        """Overwrite this method in subclass to define how to load your model from vanilla hugging face model"""
+        raise NotImplementedError(
+            "Overwrite this method in subclass to define how to load your model from vanilla hugging face model"
         )
```
Comment (michaelbenayoun) [inline optimum/modeling_base.py:310]: Why not `@abstractclassmethod`?

**Review by michaelbenayoun — APPROVED** [review]
LGTM thanks!

File: optimum/modeling_base.py (hunk 2 above)
Comment (mht-sharma) [reply optimum/modeling_base.py]: Yes makes sense, added abstract method

**Review by fxmarty — APPROVED** [review]
LGTM!

**Comment by fxmarty** [comment]
@mht-sharma the code snippet is much appreciated btw

**Pull request merged**